src/views/Export/index.tsx

```diff
@@ -0,0 +1,381 @@
+            });
+        },
+    });
+
+    const [
+        geoOptionsPending,
+        ,
+        ,
+        getGeoOptions,
+    ] = useRequest<unknown>({
+        url: 'server://geo-options/',
+        method: 'GET',
+        query: {
+            project: projectId,
+        },
+        schemaName: 'geoOptions',
+        onSuccess: (response) => {
+            setGeoOptions({ projectId, locations: response });
+        },
+        onFailure: (error) => {
+            notify.send({
+                title: _ts('export', 'geoLabel'),
+                type: notify.type.ERROR,
+                message: error,
+                duration: notify.duration.MEDIUM,
+            });
+        },
+    });
+
+    useEffect(() => {
+        getAnalysisFramework();
+        getGeoOptions();
+    },
+    [getAnalysisFramework, getGeoOptions, projectId]);
+
+    useEffect(
+        getLeads,
+        [projectId, leadsFilters],
+    );
+
+    useEffect(() => {
+        setActiveExportTypeKey('word');
+        setPreviewId(undefined);
+        setReportStructure([]);
+        setDecoupledEntries(true);
+    }, [projectId]);
+
+    useEffect(() => {
+        const textWidgetList = getTextWidgetsFromFramework(analysisFramework);
+        const contextualWidgetList = getContextualWidgetsFromFramework(analysisFramework);
+        setTextWidgets(textWidgetList);
+        setContextualWidgets(contextualWidgetList);
+    }, [analysisFramework]);
+
+    useEffect(() => {
+        const structure = createReportStructure(
+            analysisFramework,
+            reportStructureVariant,
+        );
+        setReportStructure(structure);
+    }, [analysisFramework, reportStructureVariant]);
+
+    const handleShowGroupsChange = setShowGroups;
+
+    const handleSelectLeadChange = useCallback((key: number, value: boolean) => {
+        const newLeads = produce(leads, (safeLeads) => {
+            const index = safeLeads.findIndex(d => d.id === key);
+            if (index !== -1) {
+                // eslint-disable-next-line no-param-reassign
+                safeLeads[index].selected = value;
+            }
+        });
+        setLeads(newLeads);
+    }, [leads]);
+
+    const handleSelectAllLeads = useCallback((selectAll: boolean) => {
+        const newLeads = leads.map(l => ({
+            ...l,
+            selected: selectAll,
+        }));
+
+        setLeads(newLeads);
+    }, [leads]);
+
+    const handleReportStructureChange = setReportStructure;
+
+    const handleTextWidgetsSelection = setTextWidgets;
+
+    const handleContextualWidgetsSelection = setContextualWidgets;
+
+    const handleDecoupledEntriesChange = setDecoupledEntries;
+
+    const handleExportTypeSelectButtonClick = setActiveExportTypeKey;
+
+    const handlePreview = setPreviewId;
+
+    const handleReportStructureVariantChange = useCallback((value: ReportStructureVariant) => {
+        const report = createReportStructure(analysisFramework, value);
```

Comment:
I think we should make this as reactive as possible and not calculate report here.
But use useMemo to calculate that report.